spec/editors_draft.html

```diff
@@ -3298,6 +3298,261 @@ <h2>Evaluation Semantics</h2>
+         class="nohighlight example">
+      <!--
+      _:b004 <http://example.org/p3> "[ _:b004 ]"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/List> .
+      -->
+      </pre>
+
+      <p>Loading the Turtle in <a href="#load-renaming-blank-reuse"></a> with a triple with a <a>cdt:Map literal</a> containing two map entries, both having the same blank node as their value, MUST result in a system-internal representation with a single triple whose <a>cdt:Map literal</a> object contains two map entries having the same blank node value, and whose identifier MUST NOT be "b0" (e.g., <a href="#load-renaming-blank-reuse-result"></a>).</p>
+      <pre id="load-renaming-blank-reuse"
+         title="Consistent rewriting of blank node identifiers that are used multiple times in a cdt:Map literal"
+         data-transform="updateExample"
+         data-content-type="text/turtle"
+         class="nohighlight example">
+      <!--
+      ex:s ex:p3 "{ 'this': _:b0, 'again': _:b0 }"^^cdt:Map .
+      -->
+      </pre>
+      <pre id="load-renaming-blank-reuse-result"
+         title="Consistent rewriting of blank node identifiers that are used multiple times in a cdt:Map literal: Result"
+         data-transform="updateExample"
+         data-content-type="text/n-triples"
+         class="nohighlight example">
+      <!--
+      <http://example.org/s> <http://example.org/p3> "{'this': _:b005,'again': _:b005}"^^<http://w3id.org/awslabs/neptune/SPARQL-CDTs/Map> .
+      -->
+      </pre>
+    </section>
+
+    <section>
+      <h2>Exporting Requirements</h2>
+      <p>
+        Conforming implementations MUST process <a>cdt:List literals</a> and <a>cdt:Map literals</a> during export, replacing, in their lexical form, any substring _:<var>id</var>
+        matching the <a data-cite="TURTLE#grammar-production-BLANK_NODE_LABEL">`BLANK_NODE_LABEL`</a> production of the grammar
+        with a string "_:"+<var>b</var> where <var>b</var> is a blank node identifier, such that
+      </p>
+
+      <ul>
+        <li>within the scope of the document being serialized, all occurrences of <var>id</var> used as a blank node identifier (both inside and outside of composite values) are replaced with the same <var>b</var> value, and</li>
+        <li>within the scope of the document being serialized, two distinct <var>id</var> blank node identifiers are replaced with distinct <var>b</var> values, and</li>
+        <li>
+          if the format of the document being serialized has a mechanism for explicitly identifying blank nodes outside of cdt:List and cdt:Map literals (e.g., `_:b1` syntax in N-Triples and in Turtle, `rdf:nodeID` in RDF/XML),
+          and the blank node <var>B</var> is contained in the data to be serialized both inside and outside of composite value literals,
+          then the serializer for this document format MUST serialize <var>B</var> using the blank node identifier <var>b</var> (both inside and outside of composite values)
+        </li>
+      </ul>
```

Comment:
I am copying the following three points from a now-outdated comment as they are still unresolved.

1. While I generally agree with the new third bullet point that you have added here, this point feels somewhat detached from the context in which you are listing these bullet points. I mean, the context is that there is some substring "`_:`_id_" in the lexical form of some cdt:List or cdt:Map literal to be exported, but this bullet point does not talk at all about this substring / about _id_.
2. Related to the previous point, within the new third bullet point you seem to assume systems that represent CDT literals internally based their value form (how would such a system otherwise know that it is the same blank node B that is both inside and outside of CDT literals?). So, the issue that I see here is that it is not clear how the requirement of this bullet point can apply to systems that represent CDT literals internally based on their lexical form.
3. Another potential problem regarding this new third bullet point: While the point focuses on cases in which the format of the document being serialized has a mechanism for explicitly identifying blank nodes outside of CDT literals, I wonder what a system should do for formats that do not have such a mechanism. Assume the dataset in a system contains the triple (_b_, :p, _l_) where _b_ is a blank node and _l_ is a cdt:List literal whose value (i.e., term list) contains _b_ as well. If the dataset with this triple needs to be serialized into a format that does not provide such an aforementioned mechanism, then there is no way to serialize the triple without loosing the connection between the blank node in the subject and the blank node inside the object literal, right?

Reply:
Regarding point 3, yes, you'd lose the connection between the blank nodes. I don't know of any serialization formats that lack the ability to identify the bnodes, but I was trying to write this in a way that would be accepting of domain-specific serializations that might. For example, if you enforce restrictions on the data such that bnodes never appear outside of CDTs; or if your data model only uses bnodes in constrained ways such that they would never need to be named in the serialization (e.g. using them just as nodes to model data that could entirely be captured by Turtle-like `[ :p1 [ :p2 :v ]]` syntax). Maybe that's going too far in trying to support something there's no real evidence of?

Reply:
Now that you say it, I think yes that is going to far. So, to address my third point here, I suggest to change the beginning of the third bullet point from "if the format..." to "assuming that the format...", and then perhaps add a Note that briefly mentions that we are not aware of formats in which blank nodes cannot be named and if there was one, it may lead to the problem described in my third point here.

Reply:
I have implemented the proposal of [my previous comment](https://github.com/awslabs/SPARQL-CDTs/pull/2#discussion_r1622814034) in commit https://github.com/awslabs/SPARQL-CDTs/pull/2/commits/fdbf337cd27e12e9356a03e604413daa71f0bc68

Reply:
I have implemented the proposal of [my previous comment](https://github.com/awslabs/SPARQL-CDTs/pull/2#discussion_r1622814034) in commit https://github.com/awslabs/SPARQL-CDTs/pull/2/commits/fdbf337cd27e12e9356a03e604413daa71f0bc68

This addresses point 3 above. Points 1 and 2 remain.